bindgen/templates/ObjectTemplate.cs

```diff
@@ -205,9 +227,36 @@ public override int GetHashCode() {
     {% for cons in obj.alternate_constructors() -%}
     {%- call cs::docstring(cons, 4) %}
     {%- call cs::method_throws_annotation(cons.throws_type()) %}
+    {%- if cons.is_async() %}
+    public static async Task<{{ impl_name }}> {{ cons.name()|fn_name }} ({%- call cs::arg_list_decl(cons) -%}) {
+        return await _UniFFIAsync.UniffiRustCallAsync(
+            // Get rust future
+            _UniFFILib.{{ cons.ffi_func().name()  }}({% call cs::lower_arg_list(cons) %}),
+            // Poll
+            (IntPtr future, IntPtr continuation, IntPtr data) => _UniFFILib.{{ cons.ffi_rust_future_poll(ci) }}(future, continuation, data),
+            // Complete
+            (IntPtr future, ref UniffiRustCallStatus status) => {
+                {%- if cons.return_type().is_some() %}
+                return {% endif %}_UniFFILib.{{ cons.ffi_rust_future_complete(ci) }}(future, ref status);
+            },
+            // Free
+            (IntPtr future) => _UniFFILib.{{ cons.ffi_rust_future_free(ci) }}(future),
+            // Lift
+            (result) => {{ obj|lift_fn }}(result),
+            // Error
+            {%- match cons.throws_type() %}
+            {%- when Some(e)  %}
+            {{ e|error_converter_name }}.INSTANCE
+            {%- when None %}
+            NullCallStatusErrorHandler.INSTANCE
+            {% endmatch %}
```

Comment:
Maybe this pattern can be moved to a macro? there seems to be a few places where it's pasted in